src/_guides/language/extension-methods.md

```diff
@@ -0,0 +1,246 @@
+---
+title: Extension methods
+description: Learn how to add to existing APIs.
+---
+<?code-excerpt replace="/ *\/\/\s+ignore_for_file:[^\n]+\n//g; /([A-Z]\w*)\d\b/$1/g; /\b(main)\d\b/$1/g"?>
+
+Extension methods, introduced in Dart 2.7,
+are a way to add functionality to existing libraries.
+You might not ever need to _use_ extension methods, much less _create_ them.
+But if you generate code or find yourself writing lots of wrappers,
+consider using extension methods to reduce
+the amount of boilerplate code that you need to write.
```

Comment:
To me the main value add is not less boilerplate. It's offering APIs that as as accessible as-if they had been added to the original class.